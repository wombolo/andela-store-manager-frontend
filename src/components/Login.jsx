import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { loginUser } from '../actions/authActions';

/**
 * @param {function} event
 *  @returns {JSX} jsx
 */
export class Login extends Component {
  state = {
    email: '',
    password: '',
  };

  handleSubmit = (event) => {
    event.preventDefault();
    const userData = {
      email: this.state.email,
      password: this.state.password,
    };
    this.props.loginUser(userData, this.props.history);
  };

  handleChange = (event) => {
    this.setState({ [event.target.id]: event.target.value });
  };

  /**
   * @param {function} render
   *  @returns {JSX} jsx
   */
  render() {
    return (

      <div className="container">
        <div className="row">
          <div className="col-5 m-0-auto m-t-15">

            <div className="logo">Vy Superstore</div>
            <div className="form-box">
              <h2 className="text-center">Login </h2>

              <form onSubmit={this.handleSubmit}>
                <div className="form-group">
                  <div className="col-3">
                    <label htmlFor="email">Email</label>
                  </div>
                  <div className="col-7">
                    <input className="form-control" type="email" id="email" onChange={this.handleChange} value={this.state.email}
                    />
                  </div>
                </div>

                <div className="form-group">
                  <div className="col-3">
                    <label htmlFor="password">Password</label>
                  </div>
                  <div className="col-7">
                    <input className="form-control" type="password" id="password" onChange={this.handleChange} />
                  </div>
                </div>

                <div className="form-group">
                  <div className="col-3">&nbsp;</div>
                  <div className="col-7">
                    <button
                      type='submit'
                      className="btn btn-primary"
                      disabled={this.props.auth.isLoading}>
                      {this.props.auth.isLoading ? 'Loading...' : 'Login'}
                    </button>
                    <button type="reset" className="btn btn-dark">Reset</button>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
Login.propTypes = {
  auth: PropTypes.shape({}).isRequired,
  history: PropTypes.shape({}).isRequired,
  loginUser: PropTypes.func.isRequired,
};

export const mapStateToProps = state => ({
  auth: state.auth,
});

export default connect(mapStateToProps,
  { loginUser })(withRouter(Login));
