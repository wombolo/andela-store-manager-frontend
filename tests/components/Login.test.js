import 'babel-polyfill'
import React from 'react';
import { mount, shallow } from 'enzyme';
import sinon from 'sinon';

import { Login } from '../../src/components/Login.jsx';

let wrapper;

const props = {
  loginUser: () => null,
  history:{},
  auth: {
    user: {
      profile: {
        id: 1,
        firstname: 'Akano',
        lastname: 'Peter',
        email: 'elkenzom1@gmail.com',
        role: 'admin',
        image: 'team-7.jpg',
        password: null
      },
      iat: 1551952568,
      exp: 1552038968
    },
    isAuthenticated: true,
    error: '',
    isLoading: false
  },
};


describe('Login Component', () => {
  it('should match snapshot', () => {
    wrapper = shallow(<Login {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });

  describe('Methods in Class', () => {

    beforeEach(() => {
      // store = createStore(combineReducers({}));
      wrapper = mount(<Login {...props}/>);
    });

    afterEach(() => {
      wrapper.unmount();
    });

    it('triggers handleSubmit()', () => {
      sinon.spy(wrapper.instance(), 'handleSubmit');
      wrapper.instance().handleSubmit({preventDefault: () => {}});
      expect(wrapper.instance().handleSubmit.calledOnce)
        .toEqual(true);
    });

    it('triggers handleChange() for input title', () => {
      sinon.spy(wrapper.instance(), 'handleChange');
      wrapper.instance().handleChange({target: { value: 'u' , id: 'title'} });
      expect(wrapper.instance().handleChange.calledOnce)
        .toEqual(true);
    });

  });
});

describe('Methods in Class', () => {

it('triggers handleSubmit() when not logged in', () => {
  props.auth.isLoading = true;
  wrapper = mount(<Login {...props}/>);


  sinon.spy(wrapper.instance(), 'handleSubmit');
  wrapper.instance().handleSubmit({preventDefault: () => {}});
  expect(wrapper.instance().handleSubmit.calledOnce)
    .toEqual(true);
});

});
