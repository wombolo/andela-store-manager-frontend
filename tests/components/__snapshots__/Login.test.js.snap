// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should match snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-5 m-0-auto m-t-15"
    >
      <div
        className="logo"
      >
        Vy Superstore
      </div>
      <div
        className="form-box"
      >
        <h2
          className="text-center"
        >
          Login 
        </h2>
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <div
              className="col-3"
            >
              <label
                htmlFor="email"
              >
                Email
              </label>
            </div>
            <div
              className="col-7"
            >
              <input
                className="form-control"
                id="email"
                onChange={[Function]}
                type="email"
                value=""
              />
            </div>
          </div>
          <div
            className="form-group"
          >
            <div
              className="col-3"
            >
              <label
                htmlFor="password"
              >
                Password
              </label>
            </div>
            <div
              className="col-7"
            >
              <input
                className="form-control"
                id="password"
                onChange={[Function]}
                type="password"
              />
            </div>
          </div>
          <div
            className="form-group"
          >
            <div
              className="col-3"
            >
              Â 
            </div>
            <div
              className="col-7"
            >
              <button
                className="btn btn-primary"
                disabled={false}
                type="submit"
              >
                Login
              </button>
              <button
                className="btn btn-dark"
                type="reset"
              >
                Reset
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
